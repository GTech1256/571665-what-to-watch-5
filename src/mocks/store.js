
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {NameSpace} from "../store/reducers/root-reducer";
import film from './film';
import films from './films';
import review from './review';

const FILMS_SHOWED_PER_STEP_COUNT = 8;
const FIRST_FILTER_NAME = `All Genres`;
const EMPTY_STATE_VALUE = null;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const mockStore = configureStore([thunk]);

export default mockStore({
  [NameSpace.ADJUSTMENT]: {
    showedFilmsCount: FILMS_SHOWED_PER_STEP_COUNT,
    activeFilterGenre: FIRST_FILTER_NAME,
  },
  [NameSpace.DATA]: {
    promoFilm: film,
    film,
    films,
    filters: [FIRST_FILTER_NAME, `Action`, `Comedy`],
    comments: [review],
    favoriteFilms: films,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: EMPTY_STATE_VALUE
  },
});
