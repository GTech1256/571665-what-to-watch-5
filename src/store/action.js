export const ActionType = {
  TOGGLE_GENRE_FILTER: `TOGGLE_GENRE_FILTER`,
};

export const ActionCreator = {
  toggleGenreFilter: (genre) => ({
    type: ActionType.TOGGLE_GENRE_FILTER,
    payload: genre
  }),
};
