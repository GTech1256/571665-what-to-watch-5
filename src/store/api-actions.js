import {loadFilms, loadPromoFilm, setGenres} from "./action";
import {APIRoute} from "../const";
import {getGenresByFilms} from "../bl/film";
import {adaptFilmFromServer} from "../utils/filmAdapter";

export const fetchPromoFilm = () => (dispatch, _getState, api) => (
  api.get(APIRoute.PROMO_FILM)
    .then(({data}) => dispatch(loadPromoFilm(adaptFilmFromServer(data))))
);

export const fetchFilmsList = () => (dispatch, _getState, api) => (
  api.get(APIRoute.FILMS)
    .then(({data}) => {
      const films = data.map(adaptFilmFromServer);

      dispatch(loadFilms(films));
      dispatch(setGenres(getGenresByFilms(films)));
    })
);

// export const checkAuth = () => (dispatch, _getState, api) => (
//   api.get(APIRoute.LOGIN)
//     .then(() => dispatch(requireAuthorization(AuthorizationStatus.AUTH)))
//     .catch((err) => {
//       throw err;
//     })
// );

// export const login = ({login: email, password}) => (dispatch, _getState, api) => (
//   api.post(APIRoute.LOGIN, {email, password})
//     .then(() => dispatch(requireAuthorization(AuthorizationStatus.AUTH)))
//     .then(() => dispatch(redirectToRoute(AppRoute.RESULT)))
// );
