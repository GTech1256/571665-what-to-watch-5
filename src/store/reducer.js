import {ActionType} from "./action";
import {films, genres} from "../mocks/films";
import {extend} from "../utils/extend";
import {FILM_ALL_GENRES} from "../const";
import {getFilmsByGenre} from "../bl/film";

const initialState = {
  promoFilm: films[0],
  allFilms: films,
  films,
  genre: FILM_ALL_GENRES,
  genres
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.TOGGLE_GENRE_FILTER:
      const genre = action.payload;

      return extend(state, {
        genre,
        films: getFilmsByGenre(initialState.films, genre)
      });

    default:
      return state;
  }
};

export {reducer};
