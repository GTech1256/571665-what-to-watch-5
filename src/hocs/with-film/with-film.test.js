import React from "react";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {filmType} from "../../types";
import withFilm from "./with-film.connect";
import filmMock from "../../mocks/film";
import store from "../../mocks/store";

const MockComponent = ({film}) => <p>{film.name}</p>;

MockComponent.propTypes = {
  film: PropTypes.exact(filmType).isRequired,
};

const MockComponentWrapped = withFilm(MockComponent);

it(`withFilm is rendered correctly`, () => {
  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped filmId={filmMock.id} />
    </Provider>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
