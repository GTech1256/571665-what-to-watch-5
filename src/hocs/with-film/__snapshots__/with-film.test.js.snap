// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFilm is rendered correctly 1`] = `
<p>
  The Grand Budapest Hotel
</p>
`;
